### PR TITLE
docs(readme): better readme for help implementing a storage backend

### PR DESCRIPTION
**Pull request by chilts**
This adds a better example of using this repo as well as a pointer to the mem and mysql backends for users just trying it out. A link to the DB API has also been added.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,14 +5,37 @@ Firefox Accounts DB Server
 
 ## Usage
 
+If you would like to run a FxA Auth DB Server, then you probably want to look at the
```
Comment (pdehaan) [inline README.md]: "run _an_ FxA Auth"? or maybe "run a _Firefox Accounts_ Auth"?
I dunno, English is hard.

File: README.md [hunk 2]
```diff
@@ -5,14 +5,37 @@ Firefox Accounts DB Server
...
+[Fxa Auth DB MySql](http://github.com/mozilla/fxa-auth-db-mysql/) repo (for production usage) or the
```
Comment (pdehaan) [inline README.md]: Probably want "FxA" here for consistency.

File: README.md [hunk 3]
```diff
@@ -5,14 +5,37 @@ Firefox Accounts DB Server
...
+
+## Usage
+
 ```js
-var createServer = require('fxa-auth-db-server')
-var database = require('./db') // An object the implements the DB API described below
+// Require the dbServer. This includes dbServer.createServer() and the dbServer.errors object.
+var dbServer = require('fxa-auth-db-server')
+
+// require the database engine and pass it a logger and the dbServer.errors
```
Comment (pdehaan) [inline README.md]: What is "and pass it a logger and the `dbServer.errors`" here?
I only see a lowly `require()` in the following line (but we do pass a `dbServer.errors` instance down below on L24.

File: README.md [hunk 4]
```diff
@@ -5,14 +5,37 @@ Firefox Accounts DB Server
...
+var Store = require('./db.js')
+
+// create a DB instance and pass a logger and the dbServer.errors object
+var db = new Store(console, dbServer.errors)
 
-var server = createServer(db)
+// create the database (pass a logger and the dbServer.errors object)
```
Comment (pdehaan) [inline README.md]: Again with this "logger" and dbServer.errors object. :scratches head:

**Comment by chilts** [comment]
@pdehaan very well spotted.

Aside: I'm afraid it was me doing and re-doing it a few times ... related to both the MySql and Mem backend requiring a `DB.connect()`and passing that stuff in at different times. I tried to make this example be promise independent as the other two repos have promises in them (as current examples). Am thinking that now the storage backends have the knowledge of themselves, perhaps the Mem backend doesn't need to have a `.connect()` etc ... as I said, just an aside - thanks for the review.

**Pull request merged**